### PR TITLE
Added functionality to rename entries in favorites

### PR DESCRIPTION
**Pull request by stankis**

### CODE REVIEW

File: src/ui/components/Navigation/Core/TableView/Row/Row.tsx [hunk 1]
```diff
@@ -84,6 +89,24 @@ export class Row extends React.Component<RowProps> {
...
+            Utils.isEnabledFeature(Feature.EnableFavoritesNameAliases) && place === PLACE.FAVORITES;
+
+        const nameElement = isFavoritesNameAliasesEnabled ? (
+            <FavoritesNameWithAliasItem
+                entryId={entryId}
+                name={name}
+                alias={alias ?? ''}
+                isLocked={isLocked}
+                onRenameFavorite={onRenameFavorite}
+            ></FavoritesNameWithAliasItem>
```
Comment (imsitnikov) [inline src/ui/components/Navigation/Core/TableView/Row/Row.tsx:102]: `></FavoritesNameWithAliasItem>` -> `/>` ?

**Review by imsitnikov — CHANGES_REQUESTED** [review]

File: src/ui/components/Navigation/Core/FavoritesNameWithAliasItem/FavoritesNameWithAliasItem.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+export const FavoritesNameWithAliasItem = (props: FavoritesNameWithAliasItemProps) => {
+    const {entryId, name, alias, isLocked, onRenameFavorite} = props;
+
+    const inputRef = React.useRef<HTMLInputElement>(null);
+
+    const [editMode, setEditMode] = React.useState(false);
+    const [localText, setLocalText] = React.useState('');
+
+    useEffect(() => {
```
[the remark this answers is not in the document]
Comment (stankis) [reply src/ui/components/Navigation/Core/FavoritesNameWithAliasItem/FavoritesNameWithAliasItem.tsx:25]: ok, no problem

File: src/ui/components/Navigation/Core/FavoritesNameWithAliasItem/FavoritesNameWithAliasItem.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
+import React, {useEffect} from 'react';
+
+import {ArrowRotateLeft, Lock, Pencil} from '@gravity-ui/icons';
+import {Icon, TextInput} from '@gravity-ui/uikit';
+import block from 'bem-cn-lite';
+
+const b = block('dl-core-navigation-table-view');
```
[the remark this answers is not in the document]
Comment (stankis) [reply src/ui/components/Navigation/Core/FavoritesNameWithAliasItem/FavoritesNameWithAliasItem.tsx:7]: agree

File: src/shared/schema/us/types/favorites.ts [hunk 4]
```diff
@@ -10,10 +10,22 @@ export interface AddFavoriteResponse {
     tenantId: string;
 }
 
+export interface RenameFavoriteResponse {
```
[the remark this answers is not in the document]
Comment (stankis) [reply src/shared/schema/us/types/favorites.ts:13]: I hope yes, let me see

File: src/ui/components/Navigation/Core/FavoritesNameWithAliasItem/FavoritesNameWithAliasItem.tsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+            action: ENTRY_CONTEXT_MENU_ACTION.EDIT_FAVORITES_ALIAS,
+        });
+    };
+
+    const text = alias === '' ? name : alias;
+    const isAliasVisible = alias !== '';
+
+    return (
+        <>
+            <div title={name} className={b('name-line')}>
```
Comment (imsitnikov) [inline src/ui/components/Navigation/Core/FavoritesNameWithAliasItem/FavoritesNameWithAliasItem.tsx:40]: Let's use `title={text}` for showing correct title:
image

**Review by imsitnikov — CHANGES_REQUESTED** [review]

File: src/ui/components/EntryDialogues/DialogAddFavoritesAlias/DialogAddFavoritesAlias.tsx [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+export const DialogAddFavoritesAlias: React.FC<DialogAddFavoritesAliasProps> = ({
+    entryId,
+    visible,
+    alias,
+    onClose,
+}) => {
+    const [text, setText] = React.useState<string>(alias ?? '');
+    const dispatch = useDispatch();
+
+    function onCloseDialog() {
```
[the remark this answers is not in the document]
Comment (stankis) [reply src/ui/components/EntryDialogues/DialogAddFavoritesAlias/DialogAddFavoritesAlias.tsx]: Fixed

File: src/ui/components/EntryDialogues/EntryDialogues.tsx [hunk 7]
```diff
@@ -81,6 +86,8 @@ interface MapDialoguesProps {
     [EntryDialogName.Copy]: DialogCopyEntryProps;
     [EntryDialogName.CreateFolder]: DialogCreateFolderProps;
     [EntryDialogName.Rename]: DialogRenameEntryProps;
+    [EntryDialogName.AddFavoritesAlias]: DialogAddFavoritesAliasProps;
+    [EntryDialogName.EditFavoritesAlias]: DialogAddFavoritesAliasProps;
```
[the remark this answers is not in the document]
Comment (stankis) [reply src/ui/components/EntryDialogues/EntryDialogues.tsx]: Done

File: src/ui/components/Navigation/Core/EntryContextButton/EntryContextButton.tsx [hunk 8]
```diff
@@ -5,7 +5,7 @@ import {Icon} from '@gravity-ui/uikit';
 
 import type {NavigationEntry} from '../../../../../shared/schema';
 
-type OnClickArgs = {
+export type OnClickArgs = {
```
[the remark this answers is not in the document]
Comment (stankis) [reply src/ui/components/Navigation/Core/EntryContextButton/EntryContextButton.tsx]: No, removed, I used it before

File: src/ui/components/Navigation/Core/FavoritesNameWithAliasItem/FavoritesNameWithAliasItem.tsx (hunk 5 above)
Comment (stankis) [reply src/ui/components/Navigation/Core/FavoritesNameWithAliasItem/FavoritesNameWithAliasItem.tsx]: Discussed offline, fixed

File: src/shared/modules/entry.ts [hunk 9]
```diff
@@ -21,3 +21,7 @@ export function normalizeDestination(destination = '') {
 export function isUsersFolder(key = '') {
     return key.toLowerCase() === 'users/';
 }
+
+export const isFavoritesAliasAvailable = (alias?: string | null) => {
+    return alias !== null && alias !== undefined && alias !== '';
```
Comment (jhoncool) [inline src/shared/modules/entry.ts:26]: Why is so complex?
```
return Boolean(alias)
```

File: src/shared/schema/us/actions/favorites.ts [hunk 10]
```diff
@@ -24,6 +26,11 @@ export const favoritesActions = {
         path: ({entryId}) => `${PATH_PREFIX}/favorites/${filterUrlFragment(entryId)}`,
         params: (_, headers) => ({headers}),
     }),
+    renameFavorite: createAction<RenameFavoriteResponse, RenameFavoriteArgs>({
+        method: 'POST',
+        path: ({entryId}) => `${PATH_PREFIX}/favorites/${filterUrlFragment(entryId)}/rename`,
```
Comment (jhoncool) [inline src/shared/schema/us/actions/favorites.ts:31]: You don't need to use `filterUrlFragment` already, it's implemented in gateway

File: src/ui/components/EntryDialogues/DialogEditFavoritesAlias/DialogEditFavoritesAlias.tsx [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+
+    const onResetAlias = () => {
+        renameAlias(null);
+    };
+
+    const onChangeInput = (str: string): void => {
+        setText(str);
+    };
+
+    const aliasExists = alias !== '' && alias !== null;
```
Comment (jhoncool) [inline src/ui/components/EntryDialogues/DialogEditFavoritesAlias/DialogEditFavoritesAlias.tsx:80]: Boolean(alias)

File: src/ui/components/EntryDialogues/DialogEditFavoritesAlias/DialogEditFavoritesAlias.tsx [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+    const renameAlias = async (str: string | null) => {
+        let name = str;
+
+        if (name !== null) {
+            name = name.trim();
+            if (name === '') name = null;
+        }
+
+        if (name === alias) {
+            onClose({status: EntryDialogResolveStatus.Close});
```
Comment (jhoncool) [inline src/ui/components/EntryDialogues/DialogEditFavoritesAlias/DialogEditFavoritesAlias.tsx:46]: You can reuse onCloseDialog

File: src/ui/components/EntryDialogues/DialogEditFavoritesAlias/DialogEditFavoritesAlias.tsx [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+                dispatch(
+                    showToast({
+                        title: i18n('toast_rename-favorites-alias-failed'),
+                        name: 'DialogEditFavoritesAlias',
+                        error,
+                        withReport: true,
+                    }),
+                );
+
+                onClose({status: EntryDialogResolveStatus.Close});
```
Comment (jhoncool) [inline src/ui/components/EntryDialogues/DialogEditFavoritesAlias/DialogEditFavoritesAlias.tsx:63]: You can reuse onCloseDialog

File: src/ui/components/EntryDialogues/DialogEditFavoritesAlias/DialogEditFavoritesAlias.tsx [hunk 14]
```diff
@@ -0,0 +1,127 @@
...
+    return (
+        <Dialog
+            size="s"
+            hasCloseButton={true}
+            open={visible}
+            onEnterKeyDown={onApplyDialog}
+            onClose={onCloseDialog}
+            disableFocusTrap={true}
+        >
+            <div className={b()}>
```
Comment (jhoncool) [inline src/ui/components/EntryDialogues/DialogEditFavoritesAlias/DialogEditFavoritesAlias.tsx:93]: Usually you don't need to use an extra node inside Dialog

File: src/ui/components/Navigation/Core/FavoritesNameWithAliasItem/FavoritesNameWithAliasItem.tsx [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+        <>
+            <div title={text} className={b('name-line')}>
+                <span>{text}</span>
+                {isLocked ? <Icon data={Lock} className={b('lock')} /> : null}
+            </div>
+            {isAliasVisible && (
+                <Popover placement={['right', 'left', 'bottom', 'top']} content={name}>
+                    <div
+                        className={b('row-btn', b('edit-alias-btn', {visible: isAliasVisible}))}
+                        onClick={onLabelClick}
```
Comment (jhoncool) [inline src/ui/components/Navigation/Core/FavoritesNameWithAliasItem/FavoritesNameWithAliasItem.tsx:48]: Can you check that u can change alias when an entry is locked

File: src/ui/components/EntryDialogues/DialogEditFavoritesAlias/DialogEditFavoritesAlias.tsx (hunk 14 above)
Comment (stankis) [reply src/ui/components/EntryDialogues/DialogEditFavoritesAlias/DialogEditFavoritesAlias.tsx:93]: Agree, removed

File: src/ui/components/EntryDialogues/DialogEditFavoritesAlias/DialogEditFavoritesAlias.tsx (hunk 13 above)
Comment (stankis) [reply src/ui/components/EntryDialogues/DialogEditFavoritesAlias/DialogEditFavoritesAlias.tsx:63]: fixed

File: src/ui/components/EntryDialogues/DialogEditFavoritesAlias/DialogEditFavoritesAlias.tsx (hunk 12 above)
Comment (stankis) [reply src/ui/components/EntryDialogues/DialogEditFavoritesAlias/DialogEditFavoritesAlias.tsx:46]: agree, didn't notice

File: src/ui/components/EntryDialogues/DialogEditFavoritesAlias/DialogEditFavoritesAlias.tsx (hunk 11 above)
Comment (stankis) [reply src/ui/components/EntryDialogues/DialogEditFavoritesAlias/DialogEditFavoritesAlias.tsx:80]: agree, fixed

File: src/shared/modules/entry.ts (hunk 9 above)
Comment (stankis) [reply src/shared/modules/entry.ts:26]: Good catch, alias is a string and '0' will not break logic, fixed

File: src/ui/components/Navigation/Core/FavoritesNameWithAliasItem/FavoritesNameWithAliasItem.tsx (hunk 15 above)
Comment (stankis) [reply src/ui/components/Navigation/Core/FavoritesNameWithAliasItem/FavoritesNameWithAliasItem.tsx:48]: I checked, alias cannot be changed for this case, will think about it within next task

**Review by jhoncool — APPROVED** [review]

**Review by imsitnikov — APPROVED** [review]

File: src/shared/schema/us/actions/favorites.ts (hunk 10 above)
Comment (stankis) [reply src/shared/schema/us/actions/favorites.ts]: Removed

**Pull request merged**